src/main/java/core/mate/academy/service/BulldozerProducer.java

```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Machine> {
```

Comment:
And here to use a more specific type for the generic.
`public class BulldozerProducer implements MachineProducer<Bulldozer>`